Jenkinsfile_updatecli

```diff
@@ -1,68 +1,47 @@
 def cronExpr = env.BRANCH_IS_PRIMARY ? 'H/30 * * * *' : ''
 
-pipeline {
-  agent {
-    label 'jnlp-linux-arm64'
-  }
-
-  options {
-    buildDiscarder(logRotator(numToKeepStr: '10'))
-    timeout(time: 30, unit: 'MINUTES')
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '10')),
     disableConcurrentBuilds()
-  }
-
-  triggers {
-    cron (cronExpr)
-  }
+])
+if (cronExpr) {
+    properties([pipelineTriggers([cron(cronExpr)])])
+}
 
-  environment {
-    UPDATECLI_AZURE = credentials('updatecli-azure-serviceprincipal') //needed for az login
-  }
-  stages {
-    stage('Check Configuration Update') {
-      when {
-        expression { not { env.BRANCH_IS_PRIMARY }}
-      }
-      // Run updatecli's diff on both push and pull requests (in case a configuration change breaks updatecli)
-      steps {
-        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
-          script {
-            withCredentials([
-              usernamePassword(
-                credentialsId: 'github-app-updatecli-on-jenkins-infra', //needed for updatecli
-                usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
+node('jnlp-linux-arm64') {
+    timeout(time: 30, unit: 'MINUTES') {
+        withCredentials([
+            azureServicePrincipal('updatecli-azure-serviceprincipal'), // needed for Azure login
+            usernamePassword(
+                credentialsId: 'github-app-updatecli-on-jenkins-infra', // needed for updatecli
+                usernameVariable: 'USERNAME_VALUE',
                 passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
-              )
-            ]) {
-              sh 'az login --service-principal -u "$UPDATECLI_AZURE_CLIENT_ID" -p "$UPDATECLI_AZURE_CLIENT_SECRET" -t "$UPDATECLI_AZURE_TENANT_ID"'
-              sh 'az account set -s "$UPDATECLI_AZURE_SUBSCRIPTION_ID"'
-              sh 'updatecli version'
-              sh 'updatecli diff --values ./updatecli/values.yaml --config ./updatecli/updatecli.d'
+            )
+        ]) {
+            // Preliminary Azure login steps
+            sh 'az login --service-principal -u "$AZURE_CLIENT_ID" -p "$AZURE_CLIENT_SECRET" -t "$AZURE_TENANT_ID"'
+            sh 'az account set -s "$AZURE_SUBSCRIPTION_ID"'
+
+            // Branch conditional: if not primary, run diff; otherwise, run apply
```

Comment:
```suggestion
```

Either remove this unneeded comment, or change ti to give it a meaning (such as explaining "why do we not want to run both `diff` and `apply`?")